### ISSUE TITLE
Support for Document Intelligence

### ISSUE THREAD

**Issue by JavascriptMick** [Azure/azure-sdk-for-go#23080]
### Feature Request

Any plans to support Azure Document Intelligence in the go sdk.. similar to the Typescript sdk?

```ts
    const client = new DocumentAnalysisClient(
      this.azureDocumentAnalysisEndpoint,
      new AzureKeyCredential(this.azureDocumentAnalysisKey)
    );

    const poller = await client.beginAnalyzeDocumentFromUrl(
      'prebuilt-xxx',
      imageUrl
    );
    ```

**Comment by jhendrixMSFT** [comment]
CC @sandeep-sen

**Comment by anrothMSFT** [comment]
Hey! I've let the team, but I've flagged this for @vkurpad.

**Comment by JavascriptMick** [comment]
in the meantime, I am investigating using go to hit the rest api

**Comment by JavascriptMick** [comment]
Hi @anrothMSFT @ctstone @sandeep-sen thanks for looking into this for me, a go client would be awesome but I realise this might take some time.  In the meantime, I have been trying to workaround with the rest api but I can't get the results I need and would appreciate some help.

I'm able to interact with the API and retrieve results however I was not able to use the latest v4.0: 2024-02-29-preview version of the api due to a 404 (not found) problem so I needed to fallback to the earlier v3.1: 2023-07-31 (GA) version.  I don't think this is about my region (Australia East) because the Typescript API gives me what I need.

Unfortunately, the resulting AnalyzeDocument with the v3.1 Rest API is a different format and a much degraded richness in information (see attached samples).  Im not sure if this is due to using the earlier api version, your docs for both versions point to the same sample [JSON](https://github.com/Azure-Samples/cognitive-services-quickstart-code/blob/master/rest/FormRecognizer/how-to-guide/receipt-model-output.json) document.  My Typescript result matches this sample.

If it's possible, please indicate how I can get the same document analysis with REST that I can achieve via the typescript api either by unblocking the 404(not found) issue with the v4.0 version or by some other means.

*Details*
REST
	modelId := "prebuilt-invoice"
	apiVersion := "2023-07-31"

404 issue (curl)
```
michael_r@MacBook-Pro ~ % curl -i -X POST "${AZURE_DOCUMENT_ANALYSIS_ENDPOINT}/documentintelligence/documentModels/prebuilt-receipt:analyze?api-version=2024-02-29-preview" -H "Content-Type: application/json" -H "Ocp-Apim-Subscription-Key: ${AZURE_DOCUMENT_ANALYSIS_KEY}" --data-ascii "{'urlSource': 'https://raw.githubusercontent.com/Azure-Samples/cognitive-services-REST-api-samples/master/curl/form-recognizer/rest-api/receipt.png'}"
HTTP/2 404 
content-length: 55
content-type: application/json
date: Thu, 20 Jun 2024 02:16:26 GMT

{"error":{"code":"404","message":"Resource not found"}}%   
```
[RESTAPI_v3.1_2023-07-31_GA_AnalyzeResult.json](https://github.com/user-attachments/files/15907537/RESTAPI_v3.1_2023-07-31_GA_AnalyzeResult.json)

[TSClient_AnalyzeResult.json](https://github.com/user-attachments/files/15907552/TSClient_AnalyzeResult.json)

![sample_receipt](https://github.com/Azure/azure-sdk-for-go/assets/905728/7c69e76e-7e4c-4e01-80aa-069ebe7d72de)

**Comment by Arvi89** [comment]
Yeah that would be great. Right now I use the REST API but I'd rather use a well integrated SDK ^^

**Comment by JavascriptMick** [comment]
any update on this?

**Comment by JavascriptMick** [comment]
#onthisday 
[Regan Smith ](https://en.wikipedia.org/wiki/Regan_Smith_(swimmer))breaks the [100 metres backstroke world record](https://en.wikipedia.org/wiki/World_record_progression_100_metres_backstroke) at the [US Olympic Swimming Trials](https://en.wikipedia.org/wiki/2024_United_States_Olympic_trials_(swimming)) in [Indianapolis](https://en.wikipedia.org/wiki/Indianapolis), [Indiana](https://en.wikipedia.org/wiki/Indiana), United States. [(Olympics)](https://olympics.com/en/news/regan-smith-world-record-us-swimming-olympic-trials)

**Comment by JavascriptMick** [comment]
..... yeah, I'm just gonna flip to nano banana or any other vision model
